File: wtt_main_server/wtt_main_server_data_models/Api/LoginRequest.cs
﻿using System.Text.Json.Serialization;

namespace wtt_main_server_data.Api;

#pragma warning disable CS8618

public class LoginRequest : RegistrationRequest
{
	public string? TotpCode { get; set; }

	[JsonConstructor] public LoginRequest() { }

	public LoginRequest(RegistrationRequest rr)
	{
		this.Email = rr.Email;
		this.Password = rr.Password;
	}
}

File: tmp/Auth/Requests/LoginRequest.ts
import RegistrationRequest from "./RegistrationRequest";

export default class LoginRequest extends RegistrationRequest
{
	totpCode: string | null;

	constructor(login: string, password: string, totpCode: string | null)
	{
		super(login, password);
		this.totpCode = totpCode;
	}
}